code/modules/mob/living/carbon/human/species/machine.dm

```diff
@@ -232,3 +232,14 @@
 								)
 		var/error_message = pick(error_messages)
 		to_chat(H, "<span class='boldwarning'>[error_message]</span>")
+
+/datum/species/machine/after_equip_job(datum/job/J, mob/living/carbon/human/H)
+	var/charger_pref = H.client.prefs.active_character.speciesprefs
+	var/obj/item/organ/internal/cyberimp/arm/power_cord/apc/apc_pw = new
+	var/obj/item/organ/internal/cyberimp/arm/power_cord/cell/cell_pw = new
+	if(charger_pref)
+		charger_pref = new /obj/item/organ/internal/cyberimp/arm/power_cord/apc(H)
+		cell_pw.insert(H)
+	else
+		charger_pref = new /obj/item/organ/internal/cyberimp/arm/power_cord/cell(H)
```

Comment:
You are still making 3 implants here, where you only need one. Please look at my suggested code from a while ago and try to incorporate that, I'd be glad to help you out again.